core/strings.go

```diff
@@ -0,0 +1,164 @@
+//
+// Package strings implements unicode string abstraction
```

Comment:
I think, every package should provide some usage examples.
I didn't see this usage examples for other packages, so wrote it in package docstring.